c7n/resources/ecs.py

```diff
@@ -525,3 +538,140 @@ def process_instance(self, client, cluster, instance):
+    https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-resource-ids.html
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: tag-ecs-service
+                resource: ecs-service
+                filters:
+                  - "tag:target-tag": absent
+                  - type: taggable
+                actions:
+                  - type: tag
+                    key: target-tag
+                    value: target-value
+    """
+    permissions = ('ecs:TagResource',)
+    batch_size = 1
+
+    def process_resource_set(self, client, resources, tags):
+        mid = self.manager.resource_type.id
+        for r in resources:
+            if not ecs_taggable(self.manager.resource_type, r):
+                raise PolicyExecutionError("New format ecs arn required: " + r[mid])
+
+        tags = [{'key': t['Key'], 'value': t['Value']} for t in tags]
+        for r in resources:
+            client.tag_resource(resourceArn=r[mid], tags=tags)
+
+
+@ECSCluster.action_registry.register('remove-tag')
+@TaskDefinition.action_registry.register('remove-tag')
+@Service.action_registry.register('remove-tag')
+@Task.action_registry.register('remove-tag')
+@ContainerInstance.action_registry.register('remove-tag')
+class RemoveTagEcsResource(RemoveTag):
+    """Remove tag(s) from ECS resources
+    (ecs, ecs-task-definition, ecs-service, ecs-task, ecs-container-instance)
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ecs-cluster-remove-tag
+                resource: ecs
+                filters:
+                  - "tag:BadTag": present
+                  - taggable
+                actions:
+                  - type: remove-tag
+                    tags: ["BadTag"]
+    """
+    permissions = ('ecs:UntagResource',)
+
+    def process_resource_set(self, client, resources, keys):
+        for r in resources:
+            client.untag_resource(resourceArn=r[self.id_key], tagKeys=keys)
+
+
+@ECSCluster.action_registry.register('mark-for-op')
+@TaskDefinition.action_registry.register('mark-for-op')
+@Service.action_registry.register('mark-for-op')
+@Task.action_registry.register('mark-for-op')
+@ContainerInstance.action_registry.register('mark-for-op')
+class MarkEcsResourceForOp(TagDelayedAction):
+    """Mark ECS resources for deferred action
+    (ecs, ecs-task-definition, ecs-service, ecs-task, ecs-container-instance)
+
+    Requires arns in new format for tasks, services, and container-instances.
+    https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-resource-ids.html
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: ecs-service-invalid-tag-stop
+            resource: ecs-service
+            filters:
+              - "tag:InvalidTag": present
+              - type: taggable
+            actions:
+              - type: mark-for-op
+                op: delete
+                days: 1
+    """
+
```

Comment:
I guess this syntactically works because there's a docstring under the class (?), but I would recommend putting a `pass` statement right after it to just indicate the absence of any further definition in the class

Reply:
a docstring for a class is sufficient, a `pass` statement in addition is redundant. this is fairly common style in many python codebases and styles.